src/configLoaders/ReactConfigLoader.ts

```diff
@@ -0,0 +1,35 @@
+import ConfigLoader from './ConfigLoader';
+import createReactJestConfig from '../utils/createReactJestConfig';
+import * as path from 'path';
+
+export default class ReactConfigLoader implements ConfigLoader {
+  private loader: NodeRequire;
+  private projectRoot: string;
+
+  public constructor(projectRoot: string, loader?: NodeRequire) {
+    this.loader = loader || /* istanbul ignore next */ require;
+    this.projectRoot = projectRoot;
+  }
+
+  public loadConfig(): string {
+    // Get the location of react script, this is later used to generate the Jest configuration used for React projects.
+    const reactScriptsLocation = path.join(this.loader.resolve('react-scripts/package.json'), '..');
+
+    // Create the React configuration for Jest
+    const jestConfiguration = this.createJestConfig(reactScriptsLocation);
+    
+    // Set test environment to jsdom (otherwise Jest won't run)
+    jestConfiguration.testEnvironment = 'jsdom';
+
+    // Return a stringified Jest configuration object.
+    return JSON.stringify(jestConfiguration);
+  }
+
+  private createJestConfig(reactScriptsLocation: string): any {
+    return createReactJestConfig(
```

Comment:
You exported a single `this.loader.resolve` call to a different file, in a different folder]. Why is it not a part of this file?

Reply:
This is because I want to be able to make a stub of the dependency in my unit tests. When I don't do this I cannot stub the method.